server/app/controllers/userController.js

```diff
@@ -0,0 +1,100 @@
+import UserModel from '../models/userModel';
+import { validateEmail, validatePassword } from '../services/validators';
+import { tokenForUser } from '../services/passport';
+
+export const createUser = async (req, res, next) => {
+  const { email, password, firstName, lastName } = req.body;
+
+  const errorMessages = {};
+
+  const emailTaken = await UserModel.findOne({ email });
+  if (emailTaken) {
+    errorMessages.email = 'This email is already taken.';
```

Comment:
It's pretty cool to see how we write UserControllers has evolved over the years. Like comparing this to how it was done in a CS 52 lab... nice man.